### ISSUE TITLE
Torch 2.10's `LRScheduler._update_lr` breaks LORA

### ISSUE THREAD

**Issue by allen-dc** [OpenRLHF/OpenRLHF#1225]
DeepSpeed's `_configure_optimizer` filters empty param groups out of the underlying optimizer in place (engine.py:1576), but the LR scheduler is built before `strategy.prepare()` runs and captures base_lrs from the  original group count. The length mismatch trips torch 2.10's
    [`zip(..., strict=True)`](https://github.com/pytorch/pytorch/blob/v2.10.0/torch/optim/lr_scheduler.py#L296) in `LRScheduler._update_lr` with "zip() argument 2 is longer than argument 1" at the first scheduler step.

Affects LoRA SFT: adapter param names (`lora_A.default.weight`,  `lora_B.default.weight`), when configured with weight_decay, match none of the no-decay patterns, so the no-decay group is always empty. Filtering at construction keeps the scheduler and optimizer in sync from the start.

The fix is a single line in `openrlhf/utils/deepspeed/deepspeed_utils.py`:

```
def get_optimizer_grouped_parameters(
    model,
    weight_decay,
    no_decay_name_list=["bias", "layer_norm.weight", "layernorm.weight", "norm.weight", "ln_f.weight"],
):
    optimizer_grouped_parameters = [
        {
            "params": [
                p
                for n, p in model.named_parameters()
                if (not any(nd in n for nd in no_decay_name_list) and p.requires_grad)
            ],
            "weight_decay": weight_decay,
        },
        {
            "params": [
                p
                for n, p in model.named_parameters()
                if (any(nd in n for nd in no_decay_name_list) and p.requires_grad)
            ],
            "weight_decay": 0.0,
        },
    ]

    # Drop empty groups: DeepSpeed's _configure_optimizer mutates
    # basic_optimizer.param_groups in place to remove groups with no params, but
    # the LR scheduler is constructed before that and captures base_lrs from the
    # original group count. The resulting length mismatch trips torch 2.10's
    # zip(..., strict=True) in LRScheduler._update_lr. This hits LoRA runs where
    # adapter param names (lora_A/lora_B) match none of the no-decay patterns,
    # leaving the second group empty.
    return [g for g in optimizer_grouped_parameters if g["params"]]
```

**Comment by hijkzzz** [comment]
cc said that this bug has been fixed in the latest main branch
